linq4j/src/main/java/org/apache/calcite/linq4j/tree/ConstantExpression.java

```diff
@@ -136,6 +136,10 @@ private static ExpressionWriter write(ExpressionWriter writer,
       write(writer, value, primitive2.primitiveClass);
       return writer.append(")");
     }
+    Primitive primitive3 = Primitive.ofBox(value.getClass());
+    if (Object.class.equals(type) && primitive3 != null) {
```

Comment:
doesn't primitive just mean unboxed java type ? can you call it just boxed ?

Reply:
Wanted to stay consistent with `primitive2` couple of lines above.